[Feat] Splash 기본 구현

usa4060 opened the pull request:
## 📌 개요
- AppCoordinator 생성
- Splash Module 구현
- WalWal Root에 Splash 연결 및 실행 테스트 완료
| View 제작은 하직 하지 않음.
## ✍️ 변경사항
- Splash Module이 추가되었습니다.
- App Coordinator가 추가되었습니다
## 📷 스크린샷
X
## 🛠️ **Technical Concerns(기술적 고민)**

usa4060 committed:
chore: #24 tuist template 수정

Coordinator의 기본 dependency중 Base에 대한 명칭 변경

usa4060 committed:
chore: #24 AppCoordinator Dependency 생성

usa4060 committed:
feat: #24 AppCoordinator scaffold 생성

usa4060 committed:
fix: #24 AppCoordinator scaffold 생성

오탈자 수정

usa4060 committed:
fix: #24 tuist template 수정

오탈자 수정

usa4060 committed:
fix: #24 AppCoordinator Interface 구현

기본 케이스: To Auth / To Home

usa4060 committed:
fix: #24 AppCoordinator Implement 구현

Auth와 Home이 없는 상태라 해당 플로우는 기본 대기

usa4060 committed:
feat: #24 Splash Feature 생성

usa4060 committed:
feat: #24 DependencyFactory 의존성 추가

usa4060 committed:
chore: Sample 코드 인덴트 수정

usa4060 committed:
fix: Sample Module 사용 안하는 코드 주석 처리

usa4060 committed:
feat: #24 App Dependency 수정

usa4060 committed:
chore: #24 App LaunchScreen 추가

usa4060 committed:
feat: #24 Splash에 대한 DependencyFactorey Protocol 추가 및 구현

usa4060 committed:
feat: #24 WalWal 초기 실행 단계 Splash 연결

View 제작은 아직 X

usa4060 committed:
feat: template 추가 수정

ji-yeon224 commented on WalWal/Features/Splash/SplashPresenter/Interface/Reactors/SplashReactor.swift:
```diff
@@ -0,0 +1,26 @@
+public protocol SplashReactorMutation {}
+public protocol SplashReactorState {}
+
+public protocol SplashReactor: Reactor where Action: SplashReactorAction, Mutation: SplashReactorMutation, State: SplashReactorState {
```
리액터 부분 아직 구현할 필요는 없겠지만, 어제 이야기 했던대로 Action, Mutation, State를 protocol이 아닌 구현체로 수정해두는 것이 좋을 것 같습니다~!

usa4060 replied:
앗 넵! 템플릿도 같이 수정 해둘게요~

usa4060 committed:
fix: #24 Reactor Action, State, Mutate 수정

기존에는 각 protocol을 interface에서 선언하고, implement에서 채택하여 사용하는방식.
-> interface에서 직접 Action, State, Mutate에 대한 구현부를 작성

usa4060 committed:
chore: indent 수정

usa4060 committed:
fix: #24 Tuist Reactor Template 수정

ji-yeon224 reviewed (approved):
고생하셨습니다😎

kimkyunghun3 commented on WalWal/Coordinators/SampleAuth/SampleAuthCoordinator/Implement/SampleAuthCoordinatorImpl.swift:
```diff
@@ -60,10 +60,10 @@ public final class SampleAuthCoordinatorImp: SampleAuthCoordinator {
-    let AuthMainVC = dependencyFactory.makeAuthViewController(reactor: reactor)
-    self.baseViewController = AuthMainVC
-    self.pushViewController(viewController: AuthMainVC, animated: false)
+    /// let reactor = dependencyFactory.makeAuthReactor(coordinator: self)
```
이런 부분은 TODO 같은거 써있으면 안까먹고 처리할거같아요~

kimkyunghun3 commented on WalWal/App/Sources/AppDelegate.swift:
```diff
@@ -6,16 +6,23 @@
+import DependencyFactoryImp
 
 @main
 class AppDelegate: UIResponder, UIApplicationDelegate {
```
다른곳도 final 붙여있는데 요기도 할까요~

kimkyunghun3 reviewed (approved):
고생했슴다~ 사소한 수정이라 어프했어요~

Zoe0929 reviewed (approved):
고생하셨습니다!

usa4060 committed:
fix: #24 AppDelegate 접근제한자 수정

usa4060 committed:
chore: #24 TODO 추가
